### DISCUSSION TITLE
[WindowsAppSdk][WASM] Writing byte[] to BitmapImage?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#7954] [Q&A]
I fail to understand and what to use instead of 
```
using (InMemoryRandomAccessStream stream = new InMemoryRandomAccessStream())
{
	using (DataWriter writer = new DataWriter(stream.GetOutputStreamAt(0)))
	{
		writer.WriteBytes(byteArray);
		await writer.StoreAsync();
	}

	await image.SetSourceAsync(stream);
}
```

I'm aware of [this ](https://stackoverflow.com/questions/69396324/datawriter-storeasync-not-available-on-wasm)and [this](https://github.com/unoplatform/uno/pull/3161) and came up with this

```
using (InMemoryRandomAccessStream stream = new InMemoryRandomAccessStream())
{
	using (BinaryWriter writer = new BinaryWriter(stream.AsStreamForWrite()))
	{
		writer.Write(byteArray);    // getting runtime error here in an error window coming up in VS
		writer.Flush();
	}

	await image.SetSourceAsync(stream);
}
```

... which doesn't work. I also tried a couple of variations but keep hitting the wall. 

Any pointer would be highly appreciated.

**Comment by jeromelaban** [comment]
`SetSourceAsync` only accepts known supported file formats. What is your input format? 

Also, can you explain what "does not work" means? Is it an exception, a log entry, something else ?

**Reply by DierkDroth** [reply]
@jeromelaban in the sample above writer.Write() fails (browser log attached) - it does not even make it to .SetSourceAsync()

This is the content of a JPG logo file (attached)

[localhost-1643540993694.zip](https://github.com/unoplatform/uno/files/7971084/localhost-1643540993694.zip)
[logo.zip](https://github.com/unoplatform/uno/files/7971093/logo.zip)

**Reply by jeromelaban** [reply]
Reading the log shows a `System.NotImplementedException`, most likely `InMemoryRandomAccessStream`. Use another set of APIs to do the write, like a .NET memory stream.

**Reply by DierkDroth** [reply]
Thanks @jeromelaban. Does this mean the .AsStreamForWrite() approach which you suggested on the stackoverflow link above is not the correct way?

Basically I'm trying to populate a 'BitmapImage' with 'byte[]' and AFAIK there is no way to get around a 'InMemoryRandomAccessStream'. Hence my inquiry...

**Comment by DierkDroth** [comment]
I just tried a different approach
```
using (MemoryStream stream = new MemoryStream(byteArray))
{
	BitmapImage image = new BitmapImage();
	await image.SetSourceAsync(stream);
	logoImage.Source = image;
}
```
Unfortunately this doesn't work neither as a new window in VS comes up saying 'Unable to retrieve source content (Unable to retrieve source content)'
So, I'm still dead in the water.

Also, appears the image.SetSourceAsync() signature in UNO WASM is not matching that of WindowsAppSdk.

**Reply by jeromelaban** [reply]
"Unable to retrieve source content" is not an Uno error. You'll need to provide more details about the error.

**Comment by DierkDroth** [comment]
@jeromelaban thanks for follow up.

I wished I could. That's all text which I do see in a new window which comes up in VS. There is nothing else neither in browser console nor in the VS output window. :-(
I get that error right as I'm leaving the 'using' scope above. So I'm speculating there is a disposing/lifecycle issue...

![image](https://user-images.githubusercontent.com/26179242/151827609-3fa42d49-8b3d-4253-b57c-f84be4e8ea45.png)
